### PR TITLE
Adds support for NMEA UART GPS on /dev/serial0. 

### PR DESCRIPTION
**Pull request by theGOTOguy**
Supports only NMEA UART GPS devices specifically on /dev/serial0 at 9600 baud.  If I get more devices to test with, I will add auto-detection later.

### CODE REVIEW

File: devices/uartnmeagps.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    logging.info('Publishing GPS data to remote')
+
+    result = False
+    try:
+      if not self.has_transmitted_device_info:
+        result = self._try_write_to_remote('GPS', 'Model', 'Generic UART NMEA GPS')
+        self.has_transmitted_device_info = True
+
+      if self.latitude and self.longitude and abs(self.latitude) <= 90 and abs(self.longitude) <= 180:
+        result = self._try_write_to_remote('GPS', 'latitude_degrees', self.latitude) or result
```
Comment (bensonk) [inline devices/uartnmeagps.py:110]: I confess I'm a little uncomfortable that latitude and longitude are written in separate lines. The idea that one could succeed and the other fail for some reason makes me itch. Like, it's fine, but I'm not stoked about it.

File: devices/uartnmeagps.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      raise err
+
+  # Close the port when we shut down.
+  def __del__(self):
+    if self.stream:
+      self.stream.close()
+
+  # Look that keeps latitude and longitude up-to-date.
+  # TODO:  We no longer auto-set system time from GPS time.  We should re-add that maybe.
+  # TODO:  There is other data here too.  Maybe also save that.  hMSL in particular seems interesting.
```
Comment (bensonk) [inline devices/uartnmeagps.py:65]: +1 to recording altitude.

File: devices/uartnmeagps.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+
+class UartNmeaGps(Sensor):
+  def __init__(self, remotestorage, localstorage, timesource, interval=None, send_last_known_gps=False, env_file=None, **kwargs):
+    super().__init__(remotestorage, localstorage, timesource)
+    # Stream represents our connection to the UART.
+    self.stream = None
+    self.last_error = ''
+
+    # If available, we will save last known GPS coordinates to environment variables.
```
Comment (bensonk) [inline devices/uartnmeagps.py:25]: Does this save to disk every time?

**Review by bensonk — APPROVED** [review]
LGTM! A few random thoughts, but overall everything seems copacetic.

File: devices/uartnmeagps.py (hunk 1 above)
Comment (theGOTOguy) [reply devices/uartnmeagps.py:115]: Unfortunately this is baked into my backend API architecture.  Each measurement is written independently, attached to its specific timestamp.  The good news is, if one succeeded and the other failed, that would be immediately clear to anyone inspecting or using the data.

File: devices/uartnmeagps.py (hunk 2 above)
Comment (theGOTOguy) [reply devices/uartnmeagps.py]: Done

File: devices/uartnmeagps.py (hunk 3 above)
Comment (theGOTOguy) [reply devices/uartnmeagps.py:25]: Yep.  Whenever possible, we save the last-known GPS coordinates.

**Pull request merged**